src/MacroTools/Buffs/AddSpellonCastBuff.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using WCSharp.Buffs;
+using static War3Api.Common;
+
+namespace MacroTools.Buffs
+{
+  public sealed class AddSpellOnCastBuff : PassiveBuff
```

Comment:
As discussed, this should apply a visible buff. I've told Sebas how to do it, but if anyone else picks this up the documentation for doing this is here, it's very good: https://github.com/Orden4/WCSharp/wiki/BoundBuff